File: frontend/src/app/common/dto/rehearsal.dto.ts
import {MusicianDto} from './musician.dto';
import {MetronomeDto} from './metronome.dto';

export class RehearsalDto {
  id: string;
  startDatetime: Date;
  finishDatetime: Date;
  description: string;
  creator: MusicianDto;
  members: MusicianDto[];
  metronomes: MetronomeDto[];
}

File: backend/src/main/java/com/dataart/playme/dto/RehearsalDto.java
package com.dataart.playme.dto;

import com.dataart.playme.validation.ValidRehearsal;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@ValidRehearsal
public abstract class RehearsalDto {

    private Date startDatetime;

    private Date finishDatetime;

    private String description;

    private List<String> membersId;
}
